src/main/java/com/conveyal/datatools/manager/controllers/api/LabelController.java

```diff
@@ -0,0 +1,218 @@
+import com.conveyal.datatools.manager.models.FeedSource;
+import com.conveyal.datatools.manager.models.JsonViews;
+import com.conveyal.datatools.manager.models.Label;
+import com.conveyal.datatools.manager.models.Project;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import com.conveyal.datatools.manager.utils.json.JsonManager;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jetty.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Request;
+import spark.Response;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.conveyal.datatools.common.utils.SparkUtils.getPOJOFromRequestBody;
+import static com.conveyal.datatools.common.utils.SparkUtils.logMessageAndHalt;
+import static spark.Spark.*;
+import static spark.Spark.post;
+
+
+public class LabelController {
+    private static final Logger LOG = LoggerFactory.getLogger(LabelController.class);
+    private static JsonManager<Label> json = new JsonManager<>(Label.class, JsonViews.UserInterface.class);
+
+
+    /**
+     * Spark HTTP endpoint to get a single label by ID
+     */
+    private static Label getLabel(Request req, Response res) {
+        return requestLabelById(req, Actions.VIEW);
+    }
+
+    /**
+     * Spark HTTP endpoint that handles getting all labels for a handful of use cases:
+     * - for a single project (if projectId query param provided)
+     */
+    private static Collection<Label> getAllLabels(Request req, Response res) {
+        Auth0UserProfile user = req.attribute("user");
+        String projectId = req.queryParams("projectId");
+        Project project = Persistence.projects.getById(projectId);
+        if (project == null) {
+            logMessageAndHalt(req, 400, "Must provide valid projectId query param to retrieve labels.");
+        }
+
+        boolean isProjectAdmin = user.canAdministerProject(project.id, project.organizationId);
+
+        return project.retrieveProjectLabels().stream()
+                // Only return labels user has access to
+                .filter(label -> isProjectAdmin || !label.adminOnly)
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * HTTP endpoint to create a new label
+     */
+    private static Label createLabel(Request req, Response res) throws IOException {
+        Auth0UserProfile userProfile = req.attribute("user");
+        Label newLabel = getPOJOFromRequestBody(req, Label.class);
+        validate(req, newLabel);
+        // User may not be allowed to create a new label
+        checkLabelPermissions(req, newLabel, Actions.CREATE);
+
+        try {
+            Persistence.labels.create(newLabel);
+            return newLabel;
+        } catch (Exception e) {
+            logMessageAndHalt(req, 500, "Unknown error encountered creating label", e);
```

Comment:
Be consistant with your full stops at the end of these messages. Some have it, some don't. :-)